WebDevStudios/wds-blocks
# Refactor block settings

Closes #91 

### DESCRIPTION ###
- Abstracts custom color palette control to new component, `ColorPaletteControl`.
- Abstracts background options panel to new component, `BackgroundSettingsPanel`.
- Abstracts `BackgroundSettingsPanel`'s media controls to new component, `MediaControl`.
- Adds `ContrastChecker` component to `Slide` and `Accordions` blocks to check color contrast and output warning if needed.
- Updates `Slide` component to use new panel/control components.
- Updates `Accordions` block to use custom color HOCs in `save` function and `withColors` HOC in `edit` function (fixes color selection functionality).
- Removes custom color palette from global `config.js`.

### SCREENSHOTS ###

`Slide` block showing color contrast warning when using background color:
![Screenshot_2020-09-09 Edit Post ‹ rave_RPG — WordPress(1)](https://user-images.githubusercontent.com/36422618/92636366-7594f780-f294-11ea-869e-d9b26f369b30.png)

`Accordions` block showing color selections:
![Screenshot_2020-09-09 Edit Post ‹ rave_RPG — WordPress](https://user-images.githubusercontent.com/36422618/92636370-762d8e00-f294-11ea-8a43-bfb78583dbd0.png)

### OTHER ###
- [x] Does this issue pass linting? (ESLint)

### STEPS TO VERIFY ###

1. Add `Carousel` block.
2. Customize font color and background type, switching between background color, image, video. Confirm preview and frontend update accordingly.
3. Choose similar font and background colors to trigger contrast warning.
4. Add `Accordion` block.
5. Customized font and background colors. Confirm preview and frontend update accordingly. Confirm checkmarks show on currently selected color(s).
6. Choose similar font and background colors to trigger contrast warning.

### DOCUMENTATION ###
Will this pull request require updating the WDS Blocks [wiki](https://github.com/WebDevStudios/wds-blocks/wiki)?
Documentation included in `README.md` files for new components. Further documentation to be added as per #88.
